.github/workflows/node-ci-main.yml

```diff
@@ -0,0 +1,81 @@
+name: Node CI (landed on protected branch)
+
+on:
+  push:
+    branches:
+    - main
+    - changeset-release/*
+    - feature/*
+
+# Our jobs run like this to minimize wasting resource cycles:
+#   1. Prime caches for primary configuration (ubuntu on node 14).
+#      This way the next two jobs can run in parallel but rely on this primed
+#      cache.
+#   2. Coverage
+#
+# For pushes directly to a branch, we assume a PR has been used with wider
+# checks, this just makes sure our coverage data is up-to-date.
+jobs:
+  prime_cache_primary:
```

Comment:
If this is exactly the same as in `node-ci.yml` I think it'd be good to use the new "shared actions" approach that have landed in webapp: https://github.com/Khan/webapp/tree/master/.github/actions/shared-node-cache

We've had the cache break several times in webapp because of small, subtle changes to the cache key. A shared action avoids that entirely.

Reply:
I'll give that a try.